src/lib/tcp/src/tests/util/time.rs

```diff
@@ -0,0 +1,404 @@
+//! Time-related types for our unit tests. The types represent simulated times.
+
+// this is only used in doc comments.
+#[allow(unused)]
+use crate::Dependencies;
+
+/// Like [`std::time::Instant`], but the time is arbitrary and is controlled by the
+/// [`Dependencies`].
+// time is internally represented in nanoseconds
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Instant(u64);
+
+/// Like [`std::time::Duration`], but the time is arbitrary and is controlled by the
+/// [`Dependencies`].
```

Comment:
I don't follow why we can't use `std::time::Duration` here. Unlike `std::time::Instant` we *can* create arbitrary `Duration` values, so it's probably related to another reason that applies to both that I'm not following :)

Reply:
There isn't really a good reason to not use a `std::time::Duration` here in the tests. My main reason was that I think having a separate type makes it more difficult to accidentally use a `std::time::Duration` derived from a `std::time::Instant` where we would actually want a `std::time::Duration` derived from a `tests::util::Instant`. We would be using `std::time::Duration` to represent durations in two different reference frames. For example in a unit test it might be easy to accidentally create a `std::time::Duration` from a `std::time::Instant` and then add that to a `tests::util::Instant`.

But I changed this to use `std::time::Duration` instead since it's probably not worth having this extra code until we actually find that we need it.